tests/virt_autotest/qem/live_migration_src.pm

```diff
@@ -0,0 +1,213 @@
+        make_path('ulogs');
+        copy(hashed_string("$vm_name.xml"), "ulogs/$vm_name.xml");
+        $vm_info->{install_cmd} = "$virt_insatll --name $vm_name $vm_info->{v_type} --disk path=$vm_info->{path} --location $vm_info->{location} -x autoyast=$vm_info->{autoyast} --graphics vnc,port=$vm_info->{port},listen=0.0.0.0";
+        record_info($vm_info->{install_cmd});
+    }
+}
+
+
+sub run {
+    select_console 'root-ssh';
+    $dst_ip   = get_job_autoinst_vars(get_parents->[0])->{SUT_IP};
+    $inst_src = get_test_suite_data()->{location}->{"$guest"};
+    record_info("DST: $dst_ip");
+    record_info("SUT: $sut_ip");
+    assert_script_run("if [ -d $images_path/guestip ]; then rm -rf $images_path/guestip; fi; mkdir $images_path/guestip");
+    assert_script_run("if ! [ -d $images_path/vm-configs ]; then mkdir $images_path/vm-configs; fi");
+    init_vm_info;
+    #start vm installation
+    if ($vms_to_install > 0) {
+        init_vm_profile;
+        while (my ($vm_name, $vm_info) = each %$vms) {
+            next if $vm_info->{reuse};
+            assert_script_run($vm_info->{install_cmd});
+        }
+        sleep 600;
+    }
+    #boot vms into second installation stage and verify install is done
+    my $n         = keys %$vms;
+    my $time_cont = 0;
+    until ($n == 0) {
+        sleep 120;
+        $time_cont++;
+        while (my ($vm_name, $vm_info) = each %$vms) {
+            unless ($vm_info->{ip} =~ /^\d+\.\d+\.\d+\.\d+$/) {
+                $vm_info->{ip} = script_output("if [ -f $images_path/guestip/$vm_name ]; then cat $images_path/guestip/$vm_name; fi");
+                if ($vm_info->{ip} =~ /^\d+\.\d+\.\d+\.\d+$/) {
+                    $n--;
+                } else {
+                    die("$vm_name startup time out after " . $time_cont * 2 . " minutes") if ($time_cont == 5);
+                    my $vm_state = script_output("virsh domstate --domain $vm_name");
+                    assert_script_run("virsh start $vm_name") if ($vm_state eq "shut off");
+                }
+            }
+            record_info($vm_info->{ip});
+        }
+    }
+
+    mutex_wait 'dst_ready';
+    # set up ssh login dst host without password
+    assert_script_run('if [ -d /root/.ssh ]; then rm -rf /root/.ssh; fi');
+    assert_script_run('ssh-keygen -t rsa -q -f "/root/.ssh/id_rsa" -N ""');
+    assert_script_run("ssh-keyscan $dst_ip > /root/.ssh/known_hosts");
+    assert_script_run("curl -k --user root:$testapi::password -T .ssh/id_rsa.pub sftp://$dst_ip/root/.ssh/authorized_keys  --ftp-create-dirs");
+    #        validate_script_output("ssh root\@$dst_ip hostname -f", sub { m/\Q$dst_ip\E/ });
+    assert_script_run("ssh root\@$dst_ip mount $sut_ip:$images_path $images_path") unless get_var("NFS_PATH");
+
+    # migration guest one by one and health check
+    while (my ($vm_name, $vm_info) = each %$vms) {
+        for my $opt (@{$vm_info->{migrate_opts}}) {
```

Comment:
When the former migration corrupts the guest disk or crashes the guest(by product bugs), the latter migrations will not be able to run. Suggest to add guest backup and restore before migration.

Reply:
migration actually migrates memory. Do you mean backing up the guest qcows image?

Reply:
Migration actually does more than memory migration, cpu info, register info, etc. Migration does not migrate disk, but it can corrupt the disk. We ever report such bugs -- disk is not writable after migration(on dst) and no way to boot up any more. Yes. Back up guest qcow2/raw images,  and replace the disk with backup before migration.

Reply:
Ok.